### ISSUE TITLE
The test result(megaface rankn and roc) of Circle loss is so bad.

### ISSUE THREAD

**Issue by ReverseSystem001** [cavalleria/cavaface.pytorch#42]
The test result(megaface rankn and roc) of Circle loss is so bad. I trained it by efficient-b0. but when I test it on megaface. the result is so bad. Do you ever trained it? if yes. can you provice the pretrained model? ths

**Comment by cavalleria** [comment]
I trained r100 with circle loss. how about  val dataset acc ?

**Comment by ReverseSystem001** [comment]
the training acc is always stay 0.001 like below, so strange. But the val eval looks good(lower than arcface loss).
56786 Epoch 24/24 Batch 12000/12948   Training Loss 25.2287 (22.5939) Training Prec@1 0.000 (0.001)   Training Prec@5 0.000 (0.005)
56787 ============================================================
56788 Current lr 1.2296052467588773e-05
56789 ============================================================
56790 Perform Evaluation on LFW, CFP_FP, AgeD and VGG2_FP, and Save Checkpoints...
56791 Epoch 24/24, Evaluation: LFW Acc: 0.9956666666666667, CFP_FP Acc: 0.9568571428571427, AgeDB Acc: 0.9576666666666667, VGG2_FP Acc: 0.9348000000000001

**Comment by ReverseSystem001** [comment]
The training parameters is default. At first, i wonder if it is too little training iterations. But the circle loss paper is only 182k iterations. less than 24epoch. So it seems like my wonder is wrong.

**Comment by cavalleria** [comment]
> The training parameters is default. At first, i wonder if it is too little training iterations. But the circle loss paper is only 182k iterations. less than 24epoch. So it seems like my wonder is wrong.

your LOSS_NAME  is softmax or softplus?

**Comment by ReverseSystem001** [comment]
softplus.

**Comment by cavalleria** [comment]
what's the megaface and ijbc acc ?

**Comment by ReverseSystem001** [comment]
> what's the megaface and ijbc acc ?

how to look at the megaface acc? I only use the cmc_json which generated under the megaface_24 file. and generate the cmc curve. the rank1 is too low.
by the way. when I finished run the eval code about megaface. the megaface_24 only generate two  files, namely matches_facescrub_megaface_retina_1000000_1.json and cmc---json. But when I trained the model by arcface. and evaluate the megaface. the megaface_24 includes :cmc_facescrub_megaface_retina_1000000_efficientnet_1.json  embedding  embedding_clean  matches_facescrub_megaface_retina_1000000_1.json  otherFiles, is there something wrong with the evaluation code. But I only changed the model with the same eval code.

**Comment by cavalleria** [comment]
> > what's the megaface and ijbc acc ?
> 
> how to look at the megaface acc? I only use the cmc_json which generated under the megaface_24 file. and generate the cmc curve. the rank1 is too low.
> by the way. when I finished run the eval code about megaface. the megaface_24 only generate two files, namely matches_facescrub_megaface_retina_1000000_1.json and cmc---json. But when I trained the model by arcface. and evaluate the megaface. the megaface_24 includes :cmc_facescrub_megaface_retina_1000000_efficientnet_1.json embedding embedding_clean matches_facescrub_megaface_retina_1000000_1.json otherFiles, is there something wrong with the evaluation code. But I only changed the model with the same eval code.

**Comment by cavalleria** [comment]
> > > what's the megaface and ijbc acc ?
> > 
> > 
> > how to look at the megaface acc? I only use the cmc_json which generated under the megaface_24 file. and generate the cmc curve. the rank1 is too low.
> > by the way. when I finished run the eval code about megaface. the megaface_24 only generate two files, namely matches_facescrub_megaface_retina_1000000_1.json and cmc---json. But when I trained the model by arcface. and evaluate the megaface. the megaface_24 includes :cmc_facescrub_megaface_retina_1000000_efficientnet_1.json embedding embedding_clean matches_facescrub_megaface_retina_1000000_1.json otherFiles, is there something wrong with the evaluation code. But I only changed the model with the same eval code.

the eval program will generate embeddings and otherfiles, and remove embeddings and otherfiles when finish evaluation. the result is appended in service/face.result file

**Comment by Linsongrong** [comment]
hello, Has the problem been solved？
